gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1226,6 +1229,8 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
               task.setEnableArgFile(true);
             }
 
+            task.onlyIf(t -> !isPropertyTrue(project, SKIP_GENERATE_REST_MODEL));
```

Comment:
Would this automatically skip the checkRestModel task as it depends on this?

Reply:
It will not ( I tested with --rerun-tasks)
```
> Task :datarealm-restaurants-codegen:generateRestModel SKIPPED
Skipping task ':datarealm-restaurants-codegen:generateRestModel' as task onlyIf is false.
:datarealm-restaurants-codegen:generateRestModel (Thread[Execution worker for ':' Thread 7,5,main]) completed. Took 0.0 secs.
:datarealm-restaurants-codegen:checkIdl (Thread[Execution worker for ':' Thread 7,5,main]) started.

> Task :datarealm-restaurants-codegen:checkIdl SKIPPED
Skipping task ':datarealm-restaurants-codegen:checkIdl' as task onlyIf is false.
:datarealm-restaurants-codegen:checkIdl (Thread[Execution worker for ':' Thread 7,5,main]) completed. Took 0.0 secs.
:datarealm-restaurants-codegen:checkRestModel (Thread[Execution worker for ':' Thread 7,5,main]) started.

> Task :datarealm-restaurants-codegen:checkRestModel
Custom actions are attached to task ':datarealm-restaurants-codegen:checkRestModel'.
Build cache key for task ':datarealm-restaurants-codegen:checkRestModel' is 1c367c32613257037a337f7ab4000298
Task ':datarealm-restaurants-codegen:checkRestModel' is not up-to-date because:
  Executed with '--rerun-tasks'.
Checking interface compatibility with API ...
Starting process 'command '/export/apps/jdk/JDK-1_8_0_172/bin/java''. Working directory: /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen Command: /export/apps/jdk/JDK-1_8_0_172/bin/java -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/juncwang/ws/oncall/datarealm-restaurants_trunk/build/datarealm-restaurants-codegen/checkRestModel/datarealm-restaurants-codegen-pathing.jar com.linkedin.restli.tools.snapshot.check.RestLiSnapshotCompatibilityChecker --compat backwards --report /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.reviews.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.reviews.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.entityMedias.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.entityMedias.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.ingredients.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.ingredients.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.restaurants.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.restaurants.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.members.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.members.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.menuItems.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.menuItems.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.miniMenuItems.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.miniMenuItems.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.chefs.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.chefs.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-models/src/main/snapshot/com.linkedin.datarealm.orders.snapshot.json /home/juncwang/ws/oncall/datarealm-restaurants_trunk/datarealm-restaurants-codegen/src/mainGeneratedRest/snapshot/com.linkedin.datarealm.orders.snapshot.json
Successfully started process 'command '/export/apps/jdk/JDK-1_8_0_172/bin/java''
ANTLR Tool version 4.5 used for code generation does not match the current runtime version 4.7.2ANTLR Runtime version 4.5 used for parser compilation does not match the current runtime version 4.7.2ANTLR Tool version 4.5 used for code generation does not match the current runtime version 4.7.2ANTLR Runtime version 4.5 used for parser compilation does not match the current runtime version 4.7.2[checker] [RS-COMPAT]: true
[checker] [MD-COMPAT]: true
[checker]
Stored cache entry for task ':datarealm-restaurants-codegen:checkRestModel' with cache key 1c367c32613257037a337f7ab4000298
:datarealm-restaurants-codegen:checkRestModel (Thread[Execution worker for ':' Thread 7,5,main]) completed. Took 0.618 secs.
:datarealm-restaurants-codegen:checkSnapshot (Thread[Execution worker for ':' Thread 7,5,main]) started.
```

Reply:
I think we need a flag to disable the entire chain of tasks starting from generate -> check -> publish. Otherwise, it can result in unpredictable results. Also, since this flag is needed for improving build speed, skipping these additional tasks would make it better.

Reply:
Do you mean one additional flag to actually combines SKIP_IDL_CHECK && IDL_NO_PUBLISH && SNAPSHOT_NO_PUBLISH && SKIP_GENERATE_REST_MODEL? But one can do this on his own through combination of existing flags.

Right now, there are already SKIP_IDL_CHECK, IDL_NO_PUBLISH, SNAPSHOT_NO_PUBLISH three Flags

If we want to keep the current `noPublish` flag behaving the same as before, and also add another flags, I am just afraid there will be too many flags.

Reply:
Okay, good point.